scrunch/datasets.py

```diff
@@ -1088,6 +1044,47 @@ def editor(self, _):
             'Unsupported operation on the editor property'
         )
 
+    def change_editor(self, user):
+        """
+        Change the current editor of the Crunch dataset.
+
+        :param user:
+            The email address, the crunch url or a User instance of the user
+            who should be set as the new current editor of the given dataset.
+
+        :returns: None
+        """
+        def _host_from_url(url):
+            resolved = urlsplit(url)
+            return resolved.hostname
+
+        def _to_url(email):
+            api_users = 'https://{}/api/users/'.format(
+                _host_from_url(self.resource.self)
+            )
+            user_url = None
+
+            users = self.resource.session.get(api_users).payload['index']
+
+            for url, user in six.iteritems(users):
+                if user['email'] == email:
+                    user_url = url
+                    self.resource.patch({'current_editor': url})
+                    break
+            assert user_url is not None, 'Unable to resolve user url'
+
+            return user_url
+
+        def _is_url(u):
+            return u.startswith('https://') or u.startswith('http://')
+
+        if isinstance(user, User):
```

Comment:
If you have a user class, then you should move all this to that class and have this method receive a User entity and be it. Move all this logic to `User.get_user(email_or_url)`. All this logic does not belong to a `Dataset` class

This method's body would just check if it is a User, then do `self.resource.edit(current_editor=user.url)`, and if the incoming parameter is not a user, then do `user = User.get_user(user)` and then proceed.